### PR TITLE
Work Queue Factory Weighted Capacity

### PR DESCRIPTION
**Pull request by nkremerh**

### CODE REVIEW

**Comment by dthain** [comment]
@btovar your call on this one

**Comment by nkremerh** [comment]
@btovar I still need to determine what impact, if any, the steady state would have on the weighted calculation.

File: work_queue/src/work_queue_graph_log [hunk 1]
```diff
@@ -47,6 +47,19 @@ def read_log_entries(file, fields):
     pairs = zip(idxs, fields)
     epoch = None
     count_lines = 0
+    prev_line = {}
```
Comment (btovar) [inline work_queue/src/work_queue_graph_log]: To make it easier, I would do here  prev_line = None
and below at line 90:

if prev_line is None:
    prev_line = record

File: batch_job/src/work_queue_factory.c [hunk 2]
```diff
@@ -111,7 +116,69 @@ static int count_workers_needed( struct list *masters_list, int only_waiting )
...
+			time_execute_weighted = time_execute_previous;
+			time_transfer_weighted = time_transfer_previous;
+		}
+
+		int capacity_weighted = capacity_weighted_previous;
+		if(time_transfer_weighted > 0) {
+			capacity_weighted = (int) (time_execute_weighted / time_transfer_weighted);
+		}
+
+		if(execute_delta > 0 && transfer_delta > 0) {
```
Comment (btovar) [inline batch_job/src/work_queue_factory.c]: Condition is the same as line 135, please clean up.

File: batch_job/src/work_queue_factory.c [hunk 3]
```diff
@@ -111,7 +116,69 @@ static int count_workers_needed( struct list *masters_list, int only_waiting )
...
+			capacity_weighted_previous = capacity_weighted;
+			time_execute_previous = time_execute_weighted;
+			time_transfer_previous = time_transfer_weighted;
```
Comment (btovar) [inline batch_job/src/work_queue_factory.c]: Same as else in line 140, please consolidate.

**Comment by nkremerh** [comment]
@btovar Hopefully this resolves what you noted.

**Comment by btovar** [comment]
One of the comments has not been addressed.

Lines 142 ad 143 are executed regardless of positive_deltas, as they appear again in lines 152 and 153 with the opposite logic. Remove lines 142 and 143 (with the corresponding else), and take lines 152 and 153 out of the if.

**Comment by nkremerh** [comment]
Those are separate variables with different purposes. One sets the current iteration's weighted execution and weighted transfer times while the second statement sets the previous weighted execution and transfer times (to set up for the next iteration).

**Comment by btovar** [comment]
Argh, not my day today.

**Comment by btovar** [comment]
Ready to merge if you are.

**Comment by nkremerh** [comment]
It happens, thanks! It is ready on my end.

**Pull request merged**